### PR TITLE
Support more rich previews in documents

### LINKED ISSUE #5109: Add hover cards for user mentions
Similar to what we have for document links, @mentions should show a hover card with more details. This may require some refactoring of the existing hover code to allow reuse.

### PR DESCRIPTION
**Pull request by apoorv-mishra**
Closes #5109 
~Also towards #4284 for `/api/integrations.*` requests~

- [x] Add `data-url` attribute to mentions
- [x] ~Add a foreign key to `authentications` table to describe its 1:1 relationship with `integrations` table~
- [x] Add support for Iframely integration, similar in manner to Draw.io integration

### CODE REVIEW

File: server/migrations/20230510153843-add-integration-to-authentications.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        );
+      }
+
+      await queryInterface.sequelize.query(
+        `DELETE from authentications WHERE "integrationId" is null`,
+        {
+          type: queryInterface.sequelize.QueryTypes.DELETE,
+          transaction,
+        }
+      );
```
Comment (apoorv-mishra) [inline server/migrations/20230510153843-add-integration-to-authentications.js:49]: Might be a bit risky?

File: server/migrations/20230510153843-add-integration-to-authentications.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+      await queryInterface.changeColumn(
+        "authentications",
+        "integrationId",
+        {
+          type: Sequelize.UUID,
+          allowNull: false,
+        },
+        { transaction }
+      );
```
Comment (apoorv-mishra) [inline server/migrations/20230510153843-add-integration-to-authentications.js:59]: We'll have to keep or get rid of it depending on consensus arrived at for [this](https://github.com/outline/outline/pull/5308/files#r1190245657).

File: server/migrations/20230510153843-add-integration-to-authentications.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          unique: true,
+          references: {
+            model: "integrations",
+          },
+        },
+        { transaction }
+      );
+
+      const integrations = await queryInterface.sequelize.query(
+        `SELECT id, "authenticationId" FROM integrations where "authenticationId" is not null`,
```
Comment (apoorv-mishra) [inline server/migrations/20230510153843-add-integration-to-authentications.js:22]: Not sure if this is a good idea for hosted version. Certainly not if there are _too many_ rows.

File: plugins/iframely/client/Icon.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+export default function Icon({ size = 24, fill = "currentColor" }: Props) {
+  return (
+    <svg
+      fill={fill}
+      width={size}
+      height={size}
+      viewBox="0 0 24 24"
+      version="1.1"
+    >
+      <path d="m2.58 12.1 8.66-7V0L0 10zm1.64 1.36 7 5.87v-5.09L7.16 11zm18-7.38L15.32 0v5.15l4.15 3.3zm1.57 1.29-8.51 6.94v5l11.2-9.65z" />
```
Comment (apoorv-mishra) [inline plugins/iframely/client/Icon.tsx:19]: Need the right `svg` here, in terms of size @tommoor

File: plugins/iframely/client/components/SettingRow.tsx
Comment (apoorv-mishra) [inline plugins/iframely/client/components/SettingRow.tsx:1]: This is copied as it is from `app/scenes/Settings/components`. Time to move it into `shared`?

File: plugins/iframely/client/Settings.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+  return (
+    <Scene title={t("Iframely")} icon={<SlackIcon />}>
+      <Heading>{t("Iframely")}</Heading>
+
+      <form onSubmit={formHandleSubmit(handleSubmit)}>
+        <SettingRow
+          label={t("Iframely url")}
+          name="iframelyUrl"
+          description={t("Add your Iframely url here.")}
```
Comment (apoorv-mishra) [inline plugins/iframely/client/Settings.tsx:104]: This sentence probably needs change?

File: plugins/iframely/client/Settings.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+          <Input
+            placeholder="https://iframe.ly"
+            pattern="https?://.*"
+            {...register("url")}
+          />
+        </SettingRow>
+        <SettingRow
+          label={t("Iframely key")}
+          name="key"
+          description={t("Add your Iframely key here.")}
```
Comment (apoorv-mishra) [inline plugins/iframely/client/Settings.tsx:116]: Same goes for this...?

File: app/models/Integration.ts [hunk 7]
```diff
@@ -16,6 +16,8 @@ class Integration<T = unknown> extends BaseModel {
 
   collectionId: string;
 
+  authToken?: string | null;
```
[the remark this answers is not in the document]
Comment (apoorv-mishra) [reply app/models/Integration.ts:19]: This is also needed for _saving_ the token. Should create another model for this then?

**Comment by tommoor** [comment]
Closing out, the work we want to integrate here was merged in other PR's

**Pull request closed without merging**